go.mod

```diff
@@ -3,13 +3,14 @@ module sigs.k8s.io/kubetest2
 go 1.14
 
 require (
-	github.com/google/uuid v1.1.1
+	github.com/google/uuid v1.1.2
 	github.com/kballard/go-shellquote v0.0.0-20180428030007-95032a82bc51
 	github.com/octago/sflags v0.2.0
 	github.com/pkg/errors v0.9.1
 	github.com/spf13/cobra v1.0.0
 	github.com/spf13/pflag v1.0.5
-	golang.org/x/sync v0.0.0-20190911185100-cd5d95a43a6e
+	golang.org/x/sync v0.0.0-20200625203802-6e8e738ad208
 	k8s.io/klog v1.0.0
+	k8s.io/release v0.4.2-0.20200925090312-5d424b99ed9c // TODO: stick to the master only for testing
```

Comment:
I would want to know the full repercussions here. We've previously already made efforts to minimize the dependencies especially on k8s.io since they are notoriously hard to maintain.

/cc @spiffxp @BenTheElder

Reply:
this is my biggest concern. I see imports like https://github.com/kubernetes/release/blob/89fddae9b1044b99146575ad89e18ce3b929bee2/go.mod#L33 upstream, which is kind of a nasty dependency for a project intended to be embedded in other repos.

I think we probably want avoid importing staging repos and k/k in kubetest2, I'm not sure the best path forward here though.

Reply:
maybe we can break out the relevant common package(s) somewhere?

Reply:
Hm, the kubectl dependency is used by different parts of k/release, If I run `go mod vendor` then I only see a very minimal subset of k/release actually being used for this implementation:

```
> find vendor/k8s.io -type d
vendor/k8s.io
vendor/k8s.io/release
vendor/k8s.io/release/pkg
vendor/k8s.io/release/pkg/git
vendor/k8s.io/release/pkg/http
vendor/k8s.io/release/pkg/release
vendor/k8s.io/release/pkg/release/regex
vendor/k8s.io/release/pkg/command
vendor/k8s.io/release/pkg/util
vendor/k8s.io/release/pkg/gcp
vendor/k8s.io/release/pkg/gcp/gcs
vendor/k8s.io/klog
vendor/k8s.io/utils
vendor/k8s.io/utils/pointer
vendor/k8s.io/apimachinery
vendor/k8s.io/apimachinery/pkg
vendor/k8s.io/apimachinery/pkg/util
vendor/k8s.io/apimachinery/pkg/util/validation
vendor/k8s.io/apimachinery/pkg/util/validation/field
vendor/k8s.io/apimachinery/pkg/util/sets
vendor/k8s.io/apimachinery/pkg/util/errors
vendor/k8s.io/test-infra
vendor/k8s.io/test-infra/prow
vendor/k8s.io/test-infra/prow/logrusutil
vendor/k8s.io/test-infra/prow/config
vendor/k8s.io/test-infra/prow/config/secret
vendor/k8s.io/test-infra/prow/version
```

Since we would like to re-use parts of the push logic for the anago transition it would be more complicated for us to split-up the repo… OTOH we could think about having multiple `go.mod`s inside k/release.

cc @justaugustus @tpepper @puerco

Reply:
> Hm, the kubectl dependency is used by different parts of k/release, If I run go mod vendor then I only see a very minimal subset of k/release actually being used for this implementation:

I'd want some strong guarantees that we will not have anything like this as a transitive dependency in kubetest2, we want to let people extending this repo import k8s libraries freely without conflicting with this repo.

> Since we would like to re-use parts of the push logic for the anago transition it would be more complicated for us to split-up the repo… OTOH we could think about having multiple go.mods inside k/release.

I mean, if you split it up into a util / re-usable packages repo and the repo implementing anago etc. that should be fine, similar to k/k and k8s.io/utils (which is not staged, it's imported).

I'm not sure how well multiple modules in a repo works when you want to import one module from another.
I know it works OK for modules that do not import each other.

Reply:
The culprit here is `k8s.io/kubectl/pkg/cmd/util/editor` which is pulled in by the new release notes editing flow.
If that could be extracted from k/k as a general purpose CLI editor, we could clean up the imports.

Thoughts @soltysh @pwittrock @seans3?

Reply:
I can yank out kubectl fairly easy and with that, I think we can strip out the apimachinery dependency too.

Reply:
I've removed the dependecies to both [kubectl/…/editor](kubernetes/release#1599) and the errors package in [api-machinery](kubernetes/release#1598), this problem should be gone once those merge.